bin/preprocessing/preprocessing.sh

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/sh
+
+scripts_dir=$(dirname $0)
+script_name=$(basename $0)
+tmp=$(mktemp)
+
+for jinput in "$@"
+do
+    echo "[+] Preprocessing JSON file $(basename $jinput)"
+    for script in $(ls $scripts_dir)
+    do
+        if [ $script == $script_name ]; then
+            # dont execute this script
+            continue
+        fi
+        fullpath=$scripts_dir/$script
+        # because the following issue, we need to make a temp file
+        # https://stackoverflow.com/questions/6696842/how-can-i-use-a-file-in-a-command-and-redirect-output-to-the-same-file-without-t
+        ./$fullpath $jinput > $tmp
+        exit_code=$?
+        test $exit_code -eq 0 || (echo "Error with script $(basename $script).  Exit" && exit 1)
+        # Format via jq by default
+        cat $tmp | jq > $jinput
```

Comment:
Note that I chose to use jq as a default formatter, but it wil already change some JSON file which are not currently formatted correctly. 
In my opinion, it's time we switch to a standard formatting, and with this pipeline we have no choice anymore: either the scripts do it themselves individually (with the formatting they want) or we use JQ on all outputs. I preferred to go the standard uniform way.

Reply:
Most annoying effect so far I've seen of that formatting is to change big float numbers into powers of tens `0.000000000154` to `1.25 e-9`

Reply:
Are you saying that `jq` rewrites floats in this way?

Reply:
Exactly. by default jq rewrites floats in this form if the decimal form is too long, as well it uses 4indent spaces rather than the ones used in the current json file.

Reply:
There may be other differences, but i believe JQ is following the standards for JSON so we should go along with him.